src/hooks/dialog/useprompt.tsx

```diff
@@ -36,33 +24,78 @@ export const usePrompt = () => {
   ) => {
     dialog(
       title,
-      <>
-        {typeof message === "string" ? <P>{message}</P> : message}
-        <Form
-          fields={[{ label, name: "message", required: true }]}
-          labelSubmit={labelConfirm}
-          secondaryActions={[
-            {
-              children: labelCancel,
-              variant: "secondary",
-              type: "button",
-              onClick: () => {
-                setModalProps({ open: false });
-                onCancel?.();
-              },
-            },
-          ]}
-          validateOnChange={true}
-          onSubmit={(_, { message }) => {
-            setModalProps({ open: false });
-            onConfirm(message as string);
-          }}
-        />
-      </>,
+      <PromptForm
+        message={message}
+        label={label}
+        labelConfirm={labelConfirm}
+        labelCancel={labelCancel}
+        onConfirm={onConfirm}
+        onCancel={onCancel}
+        setModalProps={setModalProps}
+      />,
       undefined,
       { allowClose: false, ...modalProps },
     );
   };
 
   return fn;
 };
+
+const PromptForm = ({
```

Comment:
Is there any reason why an additional component is introduced over using the effect directly on the hook (might work as well I think?)

I like to keep changes as minimal as possible especially considering that a new similar hook useFormDialog may be introduced (see: OAB).

Reply:
Ah I think you're correct,  were acting on the returned function instead of the hook. I sill think this could be a nice moment to merge this with useFormDialog though so lets discuss further.